apps/src/templates/teacherDashboard/SectionActionDropdown.jsx

```diff
@@ -0,0 +1,271 @@
+import React, {Component, PropTypes} from 'react';
+import color from "../../util/color";
+import {sortableSectionShape, OAuthSectionTypes} from "./shapes.jsx";
+import PopUpMenu from "@cdo/apps/lib/ui/PopUpMenu";
+import i18n from '@cdo/locale';
+import {pegasus} from "../../lib/util/urlHelpers";
+import {sectionCode,
+        sectionName,
+        removeSection,
+        toggleSectionHidden,
+        importOrUpdateRoster} from './teacherSectionsRedux';
+import {connect} from 'react-redux';
+import PrintCertificates from "./PrintCertificates";
+import FontAwesome from '../FontAwesome';
+import BaseDialog from '../BaseDialog';
+import Button from '../Button';
+import DialogFooter from "./DialogFooter";
+
+const styles = {
+  actionButton:{
+    border: '1px solid ' + color.white,
+    paddingLeft: 5,
+    paddingRight: 5,
+    paddingTop: 4,
+    paddingBottom: 4,
+    borderRadius: 5,
+    color: color.lighter_gray,
+    margin: 3
+  },
+  hoverFocus: {
+    backgroundColor: color.lighter_gray,
+    border: '1px solid ' + color.light_gray,
+    borderRadius: 5,
+    paddingTop: 4,
+    paddingBottom: 4,
+    paddingLeft: 5,
+    paddingRight: 5,
+    color: color.white,
+  },
+  xIcon: {
+    paddingRight: 5,
+  },
+  heading: {
+    borderTopWidth: 0,
+    borderBottomWidth: 1,
+    borderRightWidth: 0,
+    borderLeftWidth: 0,
+    borderStyle: 'solid',
+    borderColor: color.default_text,
+    paddingBottom: 20,
+    marginBottom: 30,
+  },
+};
+
+class SectionActionDropdown extends Component {
+  static propTypes = {
+    handleEdit: PropTypes.func,
+    sectionData: sortableSectionShape.isRequired,
+
+    //Provided by redux
+    removeSection: PropTypes.func.isRequired,
+    toggleSectionHidden: PropTypes.func.isRequired,
+    sectionCode: PropTypes.string,
+    sectionName: PropTypes.string,
+    updateRoster: PropTypes.func.isRequired,
+  };
+
+  state = {
+    deleting: false,
+    open: false,
+    canOpen: true,
+    menuTop: 0, // location of dropdown menu
+    menuLeft: 0,
+    currWindowWidth: window.innerWidth, // used to calculate location of menu on resize
+  };
+
+  // Menu open
+  open = () => {
+    this.updateMenuLocation();
+    window.addEventListener("resize", this.updateMenuLocation);
```

Comment:
Resize events can fire a ton during a window resize, so it can affect performance to do much work in response to them.  It would be good to [throttle](https://lodash.com/docs/4.17.4#throttle) the `updateMenuLocation` callback here.